pkg/apis/kops/validation/validation.go

```diff
@@ -132,7 +138,21 @@ func validateSubnet(subnet *kops.ClusterSubnetSpec, fieldPath *field.Path) field
 	return allErrs
 }
 
-func validateHook(v *kops.HookSpec, fieldPath *field.Path) field.ErrorList {
+// validateFileAssetSpec is responsible for checking a FileAssetSpec is ok
+func validateFileAssetSpec(v *kops.FileAssetSpec, fieldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if v.Name == "" {
+		allErrs = append(allErrs, field.Required(fieldPath.Child("Name"), ""))
+	}
+	if v.Content == "" {
+		allErrs = append(allErrs, field.Required(fieldPath.Child("Content"), ""))
+	}
+
+	return allErrs
```

Comment:
Would be good to validate the base64 value here as otherwise will fail in nodeup